store/store.go

```diff
@@ -531,11 +531,57 @@ func (s *SnapUbuntuStoreRepository) FindSnaps(searchTerm string, channel string,
 	return snaps, nil
 }
 
-// Updates returns the available updates for a list of snap identified by fullname with channel.
-func (s *SnapUbuntuStoreRepository) Updates(installed []string, auther Authenticator) (snaps []*snap.Info, err error) {
-	// XXX: uses obsolete end point!
+// RefreshCandidate contains information for the store about the currently
+// installed snap so that the store can decide what update we should see
+type RefreshCandidate struct {
+	SnapID   string
+	Revision snap.Revision
+	Epoch    string
+	DevMode  bool
 
-	jsonData, err := json.Marshal(map[string][]string{"name": installed})
+	// the desired channel
+	Channel string
+}
+
+// the exact bits that we need to send to the store
+type currentSnapJson struct {
+	SnapID   string `json:"snap_id"`
+	Channel  string `json:"channel"`
+	Revision int    `json:"revision,omitempty"`
+	Epoch    string `json:"epoch"`
+	DevMode  bool   `json:"devmode"`
```

Comment:
It doesn't make much sense to me particularly, because you're in fact asking to send something that is _not_ the confinement property for the snap, but whether the snap was installed in devmode or not.

That said, I don't mind much to be honest. We can very easily map the store setting to whatever the store wants us to name the setting as.

@mvo5 if the store expects this to be named as `confinement`, let's please change only in `currentSnapJson` and set it appropriately with strict or devmode depending on what we find in `candidate.DevMode`.